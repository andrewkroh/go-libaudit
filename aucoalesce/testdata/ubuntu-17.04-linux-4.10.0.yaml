---
tests:
  unlink syscall: |
    type=SYSCALL msg=audit(1493129355.852:624): arch=c000003e syscall=87 success=yes exit=0 a0=55faa178f800 a1=1 a2=1 a3=7ffd21b026d0 items=2 ppid=3309 pid=3346 auid=1001 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=pts1 ses=5 comm="vim" exe="/usr/bin/vim.basic" key="auditconfig"
    type=CWD msg=audit(1493129355.852:624): cwd="/home/andrew_kroh"
    type=PATH msg=audit(1493129355.852:624): item=0 name="/etc/audit/rules.d/" inode=271071 dev=08:01 mode=040750 ouid=0 ogid=0 rdev=00:00 nametype=PARENT
    type=PATH msg=audit(1493129355.852:624): item=1 name="/etc/audit/rules.d/.audit.rules.swp" inode=271044 dev=08:01 mode=0100640 ouid=0 ogid=0 rdev=00:00 nametype=DELETE
    type=PROCTITLE msg=audit(1493129355.852:624): proctitle=76696D002F6574632F61756469742F72756C65732E642F61756469742E72756C6573
    type=EOE msg=audit(1493129355.852:624):

  rename syscall: |
    type=SYSCALL msg=audit(1493125524.089:126): arch=c000003e syscall=82 success=yes exit=0 a0=55942f993060 a1=55942fb79090 a2=fffffffffffffeb0 a3=55942fb79090 items=4 ppid=3207 pid=3231 auid=1001 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=pts1 ses=5 comm="vim" exe="/usr/bin/vim.basic" key="auditconfig"
    type=CWD msg=audit(1493125524.089:126): cwd="/home/andrew_kroh"
    type=PATH msg=audit(1493125524.089:126): item=0 name="/etc/audit/rules.d/" inode=271071 dev=08:01 mode=040750 ouid=0 ogid=0 rdev=00:00 nametype=PARENT
    type=PATH msg=audit(1493125524.089:126): item=1 name="/etc/audit/rules.d/" inode=271071 dev=08:01 mode=040750 ouid=0 ogid=0 rdev=00:00 nametype=PARENT
    type=PATH msg=audit(1493125524.089:126): item=2 name="/etc/audit/rules.d/audit.rules" inode=271112 dev=08:01 mode=0100640 ouid=0 ogid=0 rdev=00:00 nametype=DELETE
    type=PATH msg=audit(1493125524.089:126): item=3 name="/etc/audit/rules.d/audit.rules~" inode=271112 dev=08:01 mode=0100640 ouid=0 ogid=0 rdev=00:00 nametype=CREATE
    type=PROCTITLE msg=audit(1493125524.089:126): proctitle=76696D002F6574632F61756469742F72756C65732E642F61756469742E72756C6573
    type=EOE msg=audit(1493125524.089:126):

  USER_CMD: |
    type=USER_CMD msg=audit(1493162652.304:18662): pid=9729 uid=1001 auid=1001 ses=1 msg='cwd="/home/andrew_kroh" cmd="su" terminal=pts/0 res=success'

  USER_ACCT: |
    type=USER_ACCT msg=audit(1493219821.115:21415): pid=10665 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:accounting acct="root" exe="/usr/sbin/cron" hostname=? addr=? terminal=cron res=success'

  USER_AUTH: |
    type=USER_AUTH msg=audit(1493162652.324:18669): pid=9730 uid=0 auid=1001 ses=1 msg='op=PAM:authentication acct="root" exe="/bin/su" hostname=? addr=? terminal=/dev/pts/0 res=success'

  USER_START: |
    type=USER_START msg=audit(1493219821.115:21419): pid=10665 uid=0 auid=0 ses=39 msg='op=PAM:session_open acct="root" exe="/usr/sbin/cron" hostname=? addr=? terminal=cron res=success'

  CRED_ACQ: |
    type=CRED_ACQ msg=audit(1493219821.115:21416): pid=10665 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:setcred acct="root" exe="/usr/sbin/cron" hostname=? addr=? terminal=cron res=success'

  USER_ERR: |
    type=USER_ERR msg=audit(1493219819.739:21413): pid=10663 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:bad_ident acct="?" exe="/usr/sbin/sshd" hostname=185.56.82.22 addr=185.56.82.22 terminal=ssh res=failed'

  USER_END: |
    type=USER_END msg=audit(1493216221.105:21265): pid=10631 uid=0 auid=0 ses=38 msg='op=PAM:session_close acct="root" exe="/usr/sbin/cron" hostname=? addr=? terminal=cron res=success'

  CRED_DISP: |
    type=CRED_DISP msg=audit(1493216221.101:21264): pid=10631 uid=0 auid=0 ses=38 msg='op=PAM:setcred acct="root" exe="/usr/sbin/cron" hostname=? addr=? terminal=cron res=success'

  USER_LOGIN: |
    type=USER_LOGIN msg=audit(1493216050.975:21203): pid=10617 uid=0 auid=4294967295 ses=4294967295 msg='op=login acct=28696E76616C6964207573657229 exe="/usr/sbin/sshd" hostname=? addr=31.207.47.36 terminal=sshd res=failed'

  USER_CHAUTHTOK failed: |
    type=USER_CHAUTHTOK msg=audit(1493224850.928:21878): pid=10829 uid=1002 auid=1001 ses=41 msg='op=PAM:chauthtok acct="akroh" exe="/usr/bin/passwd" hostname=? addr=? terminal=pts/0 res=failed'
