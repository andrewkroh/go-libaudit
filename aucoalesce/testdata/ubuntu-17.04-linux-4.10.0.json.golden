[
  {
    "test_name": "CRED_ACQ",
    "event": {
      "@timestamp": "2017-04-26T15:17:01.115Z",
      "sequence": 21416,
      "category": "user-login",
      "record_type": "cred_acq",
      "result": "success",
      "session": "unset",
      "actor": {
        "primary": "unset",
        "secondary": "root",
        "attrs": {
          "auid": "unset",
          "uid": "0"
        }
      },
      "action": "acquired-credentials",
      "thing": {
        "primary": "cron",
        "what": "user-session"
      },
      "how": "/usr/sbin/cron",
      "data": {
        "acct": "root",
        "exe": "/usr/sbin/cron",
        "op": "PAM:setcred",
        "pid": "10665",
        "terminal": "cron"
      }
    },
    "warnings": [
      "failed to set object secondary using keys=[addr hostname] because they were not found"
    ]
  },
  {
    "test_name": "CRED_DISP",
    "event": {
      "@timestamp": "2017-04-26T14:17:01.101Z",
      "sequence": 21264,
      "category": "user-login",
      "record_type": "cred_disp",
      "result": "success",
      "session": "38",
      "actor": {
        "primary": "0",
        "secondary": "root",
        "attrs": {
          "auid": "0",
          "uid": "0"
        }
      },
      "action": "disposed-credentials",
      "thing": {
        "primary": "cron",
        "what": "user-session"
      },
      "how": "/usr/sbin/cron",
      "data": {
        "acct": "root",
        "exe": "/usr/sbin/cron",
        "op": "PAM:setcred",
        "pid": "10631",
        "terminal": "cron"
      }
    },
    "warnings": [
      "failed to set object secondary using keys=[addr hostname] because they were not found"
    ]
  },
  {
    "test_name": "USER_ACCT",
    "event": {
      "@timestamp": "2017-04-26T15:17:01.115Z",
      "sequence": 21415,
      "category": "user-login",
      "record_type": "user_acct",
      "result": "success",
      "session": "unset",
      "actor": {
        "primary": "unset",
        "secondary": "root",
        "attrs": {
          "auid": "unset",
          "uid": "0"
        }
      },
      "action": "was-authorized",
      "thing": {
        "primary": "cron",
        "what": "user-session"
      },
      "how": "/usr/sbin/cron",
      "data": {
        "acct": "root",
        "exe": "/usr/sbin/cron",
        "op": "PAM:accounting",
        "pid": "10665",
        "terminal": "cron"
      }
    },
    "warnings": [
      "failed to set object secondary using keys=[addr hostname] because they were not found"
    ]
  },
  {
    "test_name": "USER_AUTH",
    "event": {
      "@timestamp": "2017-04-25T23:24:12.324Z",
      "sequence": 18669,
      "category": "user-login",
      "record_type": "user_auth",
      "result": "success",
      "session": "1",
      "actor": {
        "primary": "1001",
        "secondary": "root",
        "attrs": {
          "auid": "1001",
          "uid": "0"
        }
      },
      "action": "authenticated",
      "thing": {
        "primary": "/dev/pts/0",
        "what": "user-session"
      },
      "how": "/bin/su",
      "data": {
        "acct": "root",
        "exe": "/bin/su",
        "op": "PAM:authentication",
        "pid": "9730",
        "terminal": "/dev/pts/0"
      }
    },
    "warnings": [
      "failed to set object secondary using keys=[addr hostname] because they were not found"
    ]
  },
  {
    "test_name": "USER_CHAUTHTOK failed",
    "event": {
      "@timestamp": "2017-04-26T16:40:50.928Z",
      "sequence": 21878,
      "category": "user-login",
      "record_type": "user_chauthtok",
      "result": "fail",
      "session": "41",
      "actor": {
        "primary": "1001",
        "secondary": "akroh",
        "attrs": {
          "auid": "1001",
          "uid": "1002"
        }
      },
      "action": "changed-password",
      "thing": {
        "primary": "pts/0",
        "what": "user-session"
      },
      "how": "/usr/bin/passwd",
      "data": {
        "acct": "akroh",
        "exe": "/usr/bin/passwd",
        "op": "PAM:chauthtok",
        "pid": "10829",
        "terminal": "pts/0"
      }
    },
    "warnings": [
      "failed to set object secondary using keys=[addr hostname] because they were not found"
    ]
  },
  {
    "test_name": "USER_CMD",
    "event": {
      "@timestamp": "2017-04-25T23:24:12.304Z",
      "sequence": 18662,
      "category": "user-space",
      "record_type": "user_cmd",
      "result": "success",
      "session": "1",
      "actor": {
        "primary": "1001",
        "secondary": "1001",
        "attrs": {
          "auid": "1001",
          "uid": "1001"
        }
      },
      "action": "ran-command",
      "thing": {
        "primary": "su",
        "what": "process"
      },
      "data": {
        "cmd": "su",
        "cwd": "/home/andrew_kroh",
        "pid": "9729",
        "terminal": "pts/0"
      }
    }
  },
  {
    "test_name": "USER_END",
    "event": {
      "@timestamp": "2017-04-26T14:17:01.105Z",
      "sequence": 21265,
      "category": "user-login",
      "record_type": "user_end",
      "result": "success",
      "session": "38",
      "actor": {
        "primary": "0",
        "secondary": "root",
        "attrs": {
          "auid": "0",
          "uid": "0"
        }
      },
      "action": "ended-session",
      "thing": {
        "primary": "cron",
        "what": "user-session"
      },
      "how": "/usr/sbin/cron",
      "data": {
        "acct": "root",
        "exe": "/usr/sbin/cron",
        "op": "PAM:session_close",
        "pid": "10631",
        "terminal": "cron"
      }
    },
    "warnings": [
      "failed to set object secondary using keys=[addr hostname] because they were not found"
    ]
  },
  {
    "test_name": "USER_ERR",
    "event": {
      "@timestamp": "2017-04-26T15:16:59.739Z",
      "sequence": 21413,
      "category": "user-login",
      "record_type": "user_err",
      "result": "fail",
      "session": "unset",
      "actor": {
        "primary": "unset",
        "secondary": "0",
        "attrs": {
          "auid": "unset",
          "uid": "0"
        }
      },
      "action": "error",
      "thing": {
        "primary": "ssh",
        "secondary": "185.56.82.22",
        "what": "user-session"
      },
      "how": "/usr/sbin/sshd",
      "data": {
        "addr": "185.56.82.22",
        "exe": "/usr/sbin/sshd",
        "hostname": "185.56.82.22",
        "op": "PAM:bad_ident",
        "pid": "10663",
        "terminal": "ssh"
      }
    }
  },
  {
    "test_name": "USER_LOGIN",
    "event": {
      "@timestamp": "2017-04-26T14:14:10.975Z",
      "sequence": 21203,
      "category": "user-login",
      "record_type": "user_login",
      "result": "fail",
      "session": "unset",
      "actor": {
        "primary": "unset",
        "secondary": "(invalid user)",
        "attrs": {
          "auid": "unset",
          "uid": "0"
        }
      },
      "action": "logged-in",
      "thing": {
        "primary": "sshd",
        "secondary": "31.207.47.36",
        "what": "user-session"
      },
      "how": "/usr/sbin/sshd",
      "data": {
        "acct": "(invalid user)",
        "addr": "31.207.47.36",
        "exe": "/usr/sbin/sshd",
        "op": "login",
        "pid": "10617",
        "terminal": "sshd"
      }
    }
  },
  {
    "test_name": "USER_START",
    "event": {
      "@timestamp": "2017-04-26T15:17:01.115Z",
      "sequence": 21419,
      "category": "user-login",
      "record_type": "user_start",
      "result": "success",
      "session": "39",
      "actor": {
        "primary": "0",
        "secondary": "root",
        "attrs": {
          "auid": "0",
          "uid": "0"
        }
      },
      "action": "started-session",
      "thing": {
        "primary": "cron",
        "what": "user-session"
      },
      "how": "/usr/sbin/cron",
      "data": {
        "acct": "root",
        "exe": "/usr/sbin/cron",
        "op": "PAM:session_open",
        "pid": "10665",
        "terminal": "cron"
      }
    },
    "warnings": [
      "failed to set object secondary using keys=[addr hostname] because they were not found"
    ]
  },
  {
    "test_name": "rename syscall",
    "event": {
      "@timestamp": "2017-04-25T13:05:24.089Z",
      "sequence": 126,
      "category": "audit-rule",
      "record_type": "syscall",
      "result": "success",
      "session": "5",
      "actor": {
        "primary": "1001",
        "secondary": "0",
        "attrs": {
          "auid": "1001",
          "egid": "0",
          "euid": "0",
          "fsgid": "0",
          "fsuid": "0",
          "gid": "0",
          "sgid": "0",
          "suid": "0",
          "uid": "0"
        }
      },
      "action": "renamed",
      "thing": {
        "primary": "/etc/audit/rules.d/audit.rules",
        "secondary": "271112",
        "what": "file"
      },
      "how": "/usr/bin/vim.basic",
      "key": "auditconfig",
      "data": {
        "a0": "55942f993060",
        "a1": "55942fb79090",
        "a2": "fffffffffffffeb0",
        "a3": "55942fb79090",
        "arch": "x86_64",
        "comm": "vim",
        "cwd": "/home/andrew_kroh",
        "exe": "/usr/bin/vim.basic",
        "exit": "0",
        "pid": "3231",
        "ppid": "3207",
        "proctitle": "vim",
        "syscall": "rename",
        "tty": "pts1"
      },
      "paths": [
        {
          "dev": "08:01",
          "inode": "271071",
          "item": "0",
          "mode": "040750",
          "name": "/etc/audit/rules.d/",
          "nametype": "PARENT",
          "ogid": "0",
          "ouid": "0",
          "rdev": "00:00"
        },
        {
          "dev": "08:01",
          "inode": "271071",
          "item": "1",
          "mode": "040750",
          "name": "/etc/audit/rules.d/",
          "nametype": "PARENT",
          "ogid": "0",
          "ouid": "0",
          "rdev": "00:00"
        },
        {
          "dev": "08:01",
          "inode": "271112",
          "item": "2",
          "mode": "0100640",
          "name": "/etc/audit/rules.d/audit.rules",
          "nametype": "DELETE",
          "ogid": "0",
          "ouid": "0",
          "rdev": "00:00"
        },
        {
          "dev": "08:01",
          "inode": "271112",
          "item": "3",
          "mode": "0100640",
          "name": "/etc/audit/rules.d/audit.rules~",
          "nametype": "CREATE",
          "ogid": "0",
          "ouid": "0",
          "rdev": "00:00"
        }
      ]
    }
  },
  {
    "test_name": "unlink syscall",
    "event": {
      "@timestamp": "2017-04-25T14:09:15.852Z",
      "sequence": 624,
      "category": "audit-rule",
      "record_type": "syscall",
      "result": "success",
      "session": "5",
      "actor": {
        "primary": "1001",
        "secondary": "0",
        "attrs": {
          "auid": "1001",
          "egid": "0",
          "euid": "0",
          "fsgid": "0",
          "fsuid": "0",
          "gid": "0",
          "sgid": "0",
          "suid": "0",
          "uid": "0"
        }
      },
      "action": "deleted",
      "thing": {
        "primary": "/etc/audit/rules.d/.audit.rules.swp",
        "secondary": "271044",
        "what": "file"
      },
      "how": "/usr/bin/vim.basic",
      "key": "auditconfig",
      "data": {
        "a0": "55faa178f800",
        "a1": "1",
        "a2": "1",
        "a3": "7ffd21b026d0",
        "arch": "x86_64",
        "comm": "vim",
        "cwd": "/home/andrew_kroh",
        "exe": "/usr/bin/vim.basic",
        "exit": "0",
        "pid": "3346",
        "ppid": "3309",
        "proctitle": "vim",
        "syscall": "unlink",
        "tty": "pts1"
      },
      "paths": [
        {
          "dev": "08:01",
          "inode": "271071",
          "item": "0",
          "mode": "040750",
          "name": "/etc/audit/rules.d/",
          "nametype": "PARENT",
          "ogid": "0",
          "ouid": "0",
          "rdev": "00:00"
        },
        {
          "dev": "08:01",
          "inode": "271044",
          "item": "1",
          "mode": "0100640",
          "name": "/etc/audit/rules.d/.audit.rules.swp",
          "nametype": "DELETE",
          "ogid": "0",
          "ouid": "0",
          "rdev": "00:00"
        }
      ]
    }
  }
]
